x-pack/plugins/file_upload/public/components/geojson_upload_form/index_name_form.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { ChangeEvent, Component } from 'react';
+import { EuiFormRow, EuiFieldText, EuiCallOut, EuiSpacer } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { validateIndexName } from '../../util/indexing_service';
+
+export interface Props {
+  indexName?: string;
+  indexNameError?: string;
+  onIndexNameChange: (name: string, error?: string) => void;
+}
+
+interface State {
+  existingIndexNames: string[];
+}
+
+export class IndexNameForm extends Component<Props, State> {
+  state: State = {
+    existingIndexNames: [],
+  };
+
+  _onIndexNameChange = async (indexName: string) => {
+    const indexNameError = await validateIndexName(indexName);
+    this.props.onIndexNameChange(indexName, indexNameError);
+  };
+
+  _onIndexNameChangeEvent = (event: ChangeEvent<HTMLInputElement>) => {
```

Comment:
With the refactor, there is no longer a need to break _onIndexNameChange and _onIndexNameChangeEvent into separate functions. Remove _onIndexNameChangeEvent function and have _onIndexNameChange function just take event: ChangeEvent<HTMLInputElement>